src/Microsoft.ComponentDetection.Common/DependencyGraph/ComponentRecorder.cs

```diff
@@ -71,11 +68,10 @@ public IEnumerable<string> GetSkippedComponents()
             return [];
         }
 
-        return this.singleFileRecorders
-            .Select(x => x.GetSkippedComponents().Keys)
-            .SelectMany(x => x)
+        return this.singleFileRecorders.Values
+            .SelectMany(x => x.GetSkippedComponents().Keys)
             .Distinct()
-            .ToImmutableList();
+            .ToArray();
```

Comment:
I suppose that the reason for this change is because we are returning an IEnumerable, am I right?

Reply:
Yes. There's no need to return an immutable list if the expected returned type is an IEnumerable. Moreover, array have a better performance compared to ImmutableList.